plugins/vbe/fit.c

```diff
@@ -0,0 +1,302 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Library for U-Boot Flat Image Tree (FIT)
+ *
+ * This tries to avoid using glib so that it can be used in other projects more
+ * easily
+ *
+ * Copyright (C) 2022 Google LLC
+ * Written by Simon Glass <sjg@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <libfdt.h>
+#include <zlib.h>
+#include "fit.h"
+
+#define FIT_CONFIG_PATH		"/configurations"
+#define FIT_IMAGE_PATH		"/images"
+
+#define FIT_PROP_COMPATIBLE	"compatible"
+#define FIT_PROP_DATA		"data"
+#define FIT_PROP_ALGO		"algo"
+#define FIT_PROP_DATA_OFFSET	"data-offset"
+#define FIT_PROP_DATA_SIZE	"data-size"
+#define FIT_PROP_STORE_OFFSET	"store-offset"
+#define FIT_PROP_VALUE		"value"
+#define FIT_PROP_SKIP_OFFSET	"skip-offset"
+#define FIT_VERSION		"version"
+
+static const char *const fit_err[FITE_COUNT] = {
+	[FITE_BAD_HEADER]	= "Bad device tree header",
+	[FITE_NO_CONFIG_NODE]	= "Missing /configuration node",
+	[FITE_NOT_FOUND]	= "Not found",
+	[FITE_NO_IMAGES_NODE]	= "Missing /images node",
+	[FITE_MISSING_IMAGE]	= "Missing image referred to by configuration",
+	[FITE_MISSING_SIZE]	= "Missing data-size for external data",
+	[FITE_MISSING_VALUE]	= "Missing value property for hash",
+	[FITE_MISSING_ALGO]	= "Missing algo property for hash",
+	[FITE_UNKNOWN_ALGO]	= "Unknown algo name",
+	[FITE_INVALID_HASH_SIZE] = "Invalid hash value size",
+	[FITE_HASH_MISMATCH]	= "Calculated hash value does not match",
+	[FITE_NEGATIVE_OFFSET]	= "Image has negative store-offset or data-offset",
+	[FITE_DATA_OFFSET_RANGE] = "Image data-offset is out of range of data",
+	[FITE_NEGATIVE_SIZE]	= "Image data-size is a negative value",
+};
+
+static const char *const fit_algo[FIT_ALGO_COUNT] = {
+	[FIT_ALGO_CRC32]	= "crc32",
+};
+
+int fit_open(struct fit_info *fit, const void *buf, size_t size)
+{
+	int ret;
+
+	ret = fdt_check_header(buf);
+	if (ret)
+		return -FITE_BAD_HEADER;
+	fit->blob = buf;
+	fit->size = size;
+
+	return false;
```

Comment:
FALSE and TRUE pls

Reply:
Want to use libfdt style for this library